website/addons/onedrive/serializer.py

```diff
@@ -0,0 +1,97 @@
+import logging
+
+from website.addons.base.serializer import OAuthAddonSerializer
+
+from website.util import api_url_for, web_url_for
+
+logger = logging.getLogger(__name__)
+
+
+class OneDriveSerializer(OAuthAddonSerializer):
+    def credentials_owner(self, user_settings=None):
+        return user_settings.owner or self.user_settings.owner
+
+    @property
+    def addon_short_name(self):
+        return 'onedrive'
+
+    @property
+    def user_is_owner(self):
+        if self.user_settings is None or self.node_settings is None:
+            return False
+
+        user_accounts = self.user_settings.external_accounts
+        return bool(
+            (
+                self.node_settings.has_auth and
+                (self.node_settings.external_account in user_accounts)
+            ) or len(user_accounts)
+        )
+
+    @property
+    def serialized_urls(self):
+        ret = self.addon_serialized_urls
+        ret.update({'settings': web_url_for('user_addons')})
+        return ret
+
+    @property
+    def addon_serialized_urls(self):
+        node = self.node_settings.owner
+
+        return {
+            'auth': api_url_for('oauth_connect',
+                               service_name='onedrive'),
+            'importAuth': node.api_url_for('onedrive_add_user_auth'),
+            'files': node.web_url_for('collect_file_trees'),
+            'folders': node.api_url_for('onedrive_folder_list'),
+            'config': node.api_url_for('onedrive_set_config'),
+            'deauthorize': node.api_url_for('onedrive_remove_user_auth'),
+            'accounts': node.api_url_for('onedrive_get_user_settings'),
+        }
+
+    def serialize_settings(self, node_settings, current_user, client=None):
```

Comment:
Instead of duplicating most of the base behavior, please implement a `def credentials_are_valid(sellf, user_settings, client)` that does the commented-out credential checking below.